Sources/LanguageServerProtocol/SupportTypes/RegistrationOptions.swift

```diff
@@ -37,9 +28,20 @@ public struct TextDocumentRegistrationOptions: RegistrationOptions, Hashable {
     self.documentSelector = documentSelector
   }
 
-  public func encodeIntoLSPAny(dict: inout [String: LSPAny]) {
-    guard let documentSelector = documentSelector else { return }
-    dict["documentSelector"] = documentSelector.encodeToLSPAny()
+  public init?(fromLSPDictionary dictionary: [String : LSPAny]) {
+    var documentSelector: DocumentSelector?
+
+    if let value = dictionary["documentSelector"] {
+      documentSelector = DocumentSelector(fromLSPArray: value)
+    }
+
+    self.documentSelector = documentSelector
+  }
+
+  public func encodeToLSPAny() -> LSPAny {
+    guard let documentSelector = documentSelector else { return .null }
```

Comment:
Should we return an empty dictionary here so that encoding + decoding `TextDocumentRegistrationOptions` round-trips?